components/search/search.service.ts

```diff
@@ -29,13 +31,23 @@ export class SearchService<T> {
         return this._optionsLookup;
     }
 
-    public set optionsLookup(lookupFn:LookupFn<T>) {
-        this._optionsLookup = lookupFn;
+    public set optionsLookup(lookupFn:LookupFn<T, boolean>) {
+        this._optionsLookup = (query) => lookupFn(query);
+        if (lookupFn.length == 2) {
+            this._selectedLookup = (query) => lookupFn(query, true);
```

Comment:
I wanted it to be DRY, but did not realize that `writeValue` method is in the base class, so we will not be repeating this check anywhere else. I'll put it into `writeValue`, I agree.